### PR TITLE
fix(addon-docs): support nuxt context in docs canvas

### LINKED ISSUE #121: Nuxt-Addons are not being loaded correctly in Docs Canvas
There seem to be an issue when loading custom plugins with nuxt-storybook. They work like expected in the normal Canvas panel but it does not work with storybook/addons-docs.

I am using a i18n plugin which is similiar to this one: [nuxt i18n example](https://nuxtjs.org/examples/i18n) but the same problem occurs with other plugins.

Versions: 

```json
{
    "nuxt": "2.14.4",
    "@nuxtjs/storybook": "2.2.0",
    "@storybook/addon-docs": "6.0.21"
}
```

I have TypeScript enabled and I used the create-nuxt-app starter to scaffold the app.

bug-addons

Thank you in advance for your help!

Best regards
eisv

### PR DESCRIPTION
**Pull request by farnabaz**
## Types of changes

- [X] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

override `prepareForInline` in order to render stories inside docs canvas with nuxt context
close #121 

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes (if not applicable, please state why)

### CODE REVIEW

**Comment by eisverticker** [comment]
I found a second issue, which is probably related to this one. It seems like assets (fonts, scss, ...) are not being loaded when you navigate to a components docs section directly

eg. http://localhost:4000/?path=/docs/typography--preview

Currently, you have to open the Canvas section first and then select the Docs section in order for assets to be loaded

**Comment by farnabaz** [comment]
> I found a second issue, which is probably related to this one. It seems like assets (fonts, scss, ...) are not being loaded when you navigate to a components docs section directly

It seems that the storybook does not load the entry file on the situation you are explained.

File: storybook/preview.js [hunk 1]
```diff
@@ -1,7 +1,46 @@
...
\ No newline at end of file
+
+// based on: https://github.com/storybookjs/storybook/blob/master/addons/docs/src/frameworks/vue/prepareForInline.ts
+const COMPONENT = 'STORYBOOK_COMPONENT';
+const VALUES = 'STORYBOOK_VALUES';
+const prepareForInline = (storyFn, { args }) => {
+  const component = storyFn()
+  const el = React.useRef(null)
+  // FIXME: This recreates the Vue instance every time, which should be optimized
+  React.useEffect(() => {
```
Comment (pi0) [inline storybook/preview.js:16]: Is this only way to fix and listen for ? :confused:

Comment (farnabaz) [reply storybook/preview.js:16]: This is how docs addon work.
I don't see any workaround to refactor the code.

**Comment by mariomka** [comment]
> I found a second issue, which is probably related to this one. It seems like assets (fonts, scss, ...) are not being loaded when you navigate to a components docs section directly
> 
> eg. http://localhost:4000/?path=/docs/typography--preview
> 
> Currently, you have to open the Canvas section first and then select the Docs section in order for assets to be loaded

Is there any workaround about this?

Thank you!

**Comment by farnabaz** [comment]
@mariomka Do you have a problem using the latest version? If so please provide a reproduction sample for it.

**Comment by mariomka** [comment]
@farnabaz Yes, I have a problem using the last version `3.0.0`. I found a workaround, import `scss` file directly into the `preview.js` file and add fonts manually in `preview-head.html`, it's not the best solution but it works.

I hope I have time this weekend to provide a reproduction.

**Comment by mariomka** [comment]
@farnabaz I found the root of the problem, it happens when you add custom configuration files (`main.js`, and `preview.js`).

Reproduction repo: https://github.com/mariomka/nuxt-storybook-docs-assets-issue

Just clone, install yarn dependencies, run storybook, and visit http://localhost:3003/?path=/docs/customtext--text directly, then go to the Canvas tab.

You can checkout the previous commit to seeing how it works well without custom configuration files.

**Comment by mariomka** [comment]
@farnabaz have you been able to take a look?

**Comment by farnabaz** [comment]
@mariomka Sorry for the late response and thank you for the reproduction repo.
This will be fixed by https://github.com/nuxt-community/storybook/commit/8b733f417a06f52a0101adb0d09988040aeccf18.
I will publish next release ASAP

**Comment by mariomka** [comment]
Thank you @farnabaz! Today I've was updating it and trying it, and now styles load well but fonts still don't load until you open the canvas tab. I've updated the reproduction repo with the new version.

**Comment by farnabaz** [comment]
@mariomka Sorry for the late response. Could you create an issue for it and a sample reproduction project?

**Pull request merged**